homeassistant/components/upnp/sensor.py

```diff
@@ -89,52 +76,36 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the UPnP/IGD sensors."""
-    udn = config_entry.data[CONFIG_ENTRY_UDN]
-    device: Device = hass.data[DOMAIN][DOMAIN_DEVICES][udn]
-
-    update_interval_sec = config_entry.options.get(
-        CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
-    )
-    update_interval = timedelta(seconds=update_interval_sec)
-    _LOGGER.debug("update_interval: %s", update_interval)
-    _LOGGER.debug("Adding sensors")
-    coordinator = DataUpdateCoordinator[Mapping[str, Any]](
-        hass,
-        _LOGGER,
-        name=device.name,
-        update_method=device.async_get_traffic_data,
-        update_interval=update_interval,
-    )
-    device.coordinator = coordinator
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
 
-    await coordinator.async_refresh()
+    LOGGER.debug("Adding sensors")
 
     sensors = [
-        RawUpnpSensor(coordinator, device, SENSOR_TYPES[BYTES_RECEIVED]),
-        RawUpnpSensor(coordinator, device, SENSOR_TYPES[BYTES_SENT]),
-        RawUpnpSensor(coordinator, device, SENSOR_TYPES[PACKETS_RECEIVED]),
-        RawUpnpSensor(coordinator, device, SENSOR_TYPES[PACKETS_SENT]),
-        DerivedUpnpSensor(coordinator, device, SENSOR_TYPES[BYTES_RECEIVED]),
-        DerivedUpnpSensor(coordinator, device, SENSOR_TYPES[BYTES_SENT]),
-        DerivedUpnpSensor(coordinator, device, SENSOR_TYPES[PACKETS_RECEIVED]),
-        DerivedUpnpSensor(coordinator, device, SENSOR_TYPES[PACKETS_SENT]),
+        RawUpnpSensor(coordinator, SENSOR_TYPES[BYTES_RECEIVED]),
+        RawUpnpSensor(coordinator, SENSOR_TYPES[BYTES_SENT]),
+        RawUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_RECEIVED]),
+        RawUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_SENT]),
+        DerivedUpnpSensor(coordinator, SENSOR_TYPES[BYTES_RECEIVED]),
+        DerivedUpnpSensor(coordinator, SENSOR_TYPES[BYTES_SENT]),
+        DerivedUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_RECEIVED]),
+        DerivedUpnpSensor(coordinator, SENSOR_TYPES[PACKETS_SENT]),
     ]
-    async_add_entities(sensors, True)
+    async_add_entities(sensors)
 
 
-class UpnpSensor(CoordinatorEntity, SensorEntity):
+class UpnpSensor(UpnpEntity, SensorEntity):
```

Comment:
Shouldn't this sensor have a `DEVICE_CLASS`, like the binary sensor has? Though I don't see an existing device class which fits in `homeassistant.components.sensor.DEVICE_CLASSES`.

Reply:
Device classes can be used where they fit but is not needed, and since no devices classes make sense we should not try to force one.
And this base also provides an icon, which entities that has a device class should not define.

Reply:
Alright.